### PR TITLE
github workflows : run TPM related tests on a TPM enabled vm

### PR DESCRIPTION
**Pull request by shjala**

### CODE REVIEW

**Comment by shjala** [comment]
Close it for now

**Comment by shjala** [comment]
This has some impact on the running time of the `unit` workflow (adds ~ 10 min), It is possible to make the TPM tests run only when a file in the `/pkg/pillar/evetpm` is changed, but I think that defeats the purpose of the testing because some change in `evetpm` dependencies might have an impact on it.

File: tools/tpm-tests-on-qemu.sh [hunk 1]
```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env bash
+#
+# Copyright (c) 2023 Zededa, Inc.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Some of the EVE unit-tests require a TPM to run, on Github runners there is
+# no TPM, and as a result, we skip those tests. It is possible to simulate a TPM
+# on the host using the tpm_vtpm_proxy kernel module and swtp, but there is a good
```
Comment (eriknordmark) [inline tools/tpm-tests-on-qemu.sh:8]: A yetus/blanks issue to fix on this line.

Comment (shjala) [reply tools/tpm-tests-on-qemu.sh]: fixed.

**Comment by shjala** [comment]
@eriknordmark yes I did, it worked in [act](https://github.com/nektos/act) (for testing github actions locally) and on github too, results of the last run [here](https://github.com/lf-edge/eve/actions/runs/5854048776/job/15869217931).

**Review by eriknordmark — APPROVED** [review]
LGTM

**Pull request merged**